File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.java
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public int solution(int bridge_length, int weight, int[] truck_weights) {
        int answer = 0;

        Queue<Integer> bridge = new LinkedList<>();

        for (int i = 0; i < bridge_length; i++) {
            bridge.offer(0);
        }
        int bridge_cur_weight = 0;
        int idx = 0;


        while (!bridge.isEmpty()) {
            answer++;
            bridge_cur_weight -= bridge.poll();

            if (idx < truck_weights.length) {
                if (bridge_cur_weight + truck_weights[idx] <= weight) {
                    bridge.offer(truck_weights[idx]);
                    bridge_cur_weight += truck_weights[idx];
                    idx++;
                } else {
                    bridge.offer(0);
                }
            }
        }

        return answer;
    }
}

File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.py
from collections import deque

def solution(bridge_length, weight, truck_weights):
    answer = 0

    bridge = deque([0] * bridge_length)
    bridge_cur_weight = 0
    idx = 0

    while bridge:
        answer += 1
        bridge_cur_weight -= bridge.popleft()

        if idx < len(truck_weights):
            if bridge_cur_weight + truck_weights[idx] <= weight:
                bridge.append(truck_weights[idx])
                bridge_cur_weight += truck_weights[idx]
                idx += 1
            else:
                bridge.append(0)

    return answer